next.config.js

```diff
@@ -11,83 +10,41 @@ const envConfig = require('./config')
-        modules: {
-          localIdentName: '[name]-[local]-[hash:base64:5]',
-        },
-        sourceMap: true,
-        importLoaders: 1,
-        localsConvention: 'camelCase',
-      },
-    }
-
-    if (!isServer) {
-      config.optimization.splitChunks.cacheGroups.styles = {
-        name: 'styles',
-        test: new RegExp(`\\.css$`),
-        chunks: 'all',
-        enforce: true,
-        priority: 10,
-      }
-
-      config.optimization.splitChunks.cacheGroups.vendors = {
-        test: /[\\/]node_modules[\\/]/,
-        name: 'vendors',
-        chunks: 'all',
-        enforce: true,
-        priority: 20,
-      }
-    }
-
-    if (!isServer) {
-      config.plugins.push(
-        new ExtractCssChunks({
-          filename: dev
-            ? 'static/chunks/[name].css'
-            : 'static/chunks/[name].[contenthash:8].css',
-          chunkFilename: dev
-            ? 'static/chunks/[name].chunk.css'
-            : 'static/chunks/[name].[contenthash:8].chunk.css',
-        })
+  webpack(config) {
+    const { rules } = config.module
+
+    // TODO: Remove once https://github.com/zeit/next.js/issues/10584 is solved and released
+    // Find the array of "style rules" in the webpack config.
+    // This is the array of webpack rules that:
+    // - is inside a 'oneOf' block
+    // - contains a rule that matches 'file.css'
+    const styleRules = (
+      rules.find(
+        m => m.oneOf && m.oneOf.find(({ test: reg }) => reg.test('file.css'))
+      ) || {}
+    ).oneOf
+    if (!styleRules) return config
+    // Find all the webpack rules that handle CSS modules
+    // Look for rules that match '.module.css'
+    // but aren't being used to generate
+    // error messages.
+    const cssModuleRules = [
+      styleRules.find(
+        ({ test: reg, use }) =>
+          reg.test('file.module.css') && use.loader !== 'error-loader'
+      ),
+    ].filter(n => n) // remove 'undefined' values
+    // Add the 'localsConvention' config option to the CSS loader config
+    // in each of these rules.
+    cssModuleRules.forEach(cmr => {
+      // Find the item inside the 'use' list that defines css-loader
+      const cssLoaderConfig = cmr.use.find(({ loader }) =>
+        loader.includes('css-loader')
       )
-    }
-
-    const hotCss = !isServer && {
-      loader: ExtractCssChunks.loader,
-      options: {
-        hot: dev,
-      },
-    }
-
-    config.module.rules.push({
-      test: /\.css$/,
-      exclude: /\/design\/lib/,
-      use: [hotCss, cssLoader, postcssLoader].filter(Boolean),
-    })
-
-    config.module.rules.push({
-      test: /\.css$/,
-      include: /\/design\/lib/,
-      use: [
-        hotCss,
-        { ...cssLoader, options: { ...cssLoader.options, modules: false } },
-        postcssLoader,
-      ].filter(Boolean),
+      if (cssLoaderConfig && cssLoaderConfig.options) {
+        // Patch it with the new config
+        cssLoaderConfig.options.localsConvention = 'camelCase'
```

Comment:
Why could not you pass an option to Next's `withCss`?

Reply:
Because `withCss` requires plugin. This solution doesn't require any additional packages. It's default one from [Next.js](https://nextjs.org/docs/advanced-features/customizing-postcss-config)